### PR TITLE
Add new test to robot name to prevent modification of internal state

### PR DESCRIPTION
**Pull request by matthewmorgan**
As described in #140.  This also required changing the solution, which returned the actual variable containing the internal state.

### CODE REVIEW

File: robot-name/example.js [hunk 1]
```diff
@@ -1,47 +1,27 @@
...
+  }
+
+  get name() {
+    return this.robotName;
   }
 
   reset() {
-    this.name = generateName();
+    this.robotName = generateName();
+    return this;
```
Comment (rchavarria) [inline robot-name/example.js]: Is `reset()` required to return `this`? If specs don't use the returned value, why not just simple return nothing?

File: robot-name/example.js [hunk 2]
```diff
@@ -1,47 +1,27 @@
...
-}
-
-function getNotGeneratedName(name) {
-  if (generatedNames[name]) {
-    return generateName();
-  }
-
-  generatedNames[name] = true;
+const generateName = () => {
+  let name = alpha.charAt(random(26))
```
Comment (rchavarria) [inline robot-name/example.js]: Should we extract the magic number `26` to a constant calculated from `alpha`'s length?

File: robot-name/example.js [hunk 3]
```diff
@@ -1,47 +1,27 @@
...
+    +alpha.charAt(random(26))
```
Comment (rchavarria) [inline robot-name/example.js]: Should we extract `10` to a constant?

File: robot-name/example.js (hunk 2 above)
Comment (matthewmorgan) [inline robot-name/example.js]: Yes.

File: robot-name/example.js (hunk 1 above)
Comment (matthewmorgan) [inline robot-name/example.js]: Good point.  I made that change too.

**Comment by matthewmorgan** [comment]
Forgot to change '10'.

**Comment by rchavarria** [comment]
Yep! I was reviewing it. I'll let you change it.

**Comment by rchavarria** [comment]
Great!

**Pull request merged**